crates/trie/trie/src/witness.rs

```diff
@@ -283,73 +225,138 @@ where
         }
         Ok(proof_targets)
     }
+}
+
+/// Decodes and unrolls all nodes from the proof. Returns only sibling nodes
+/// in the path of the target and the final leaf node with updated value.
+pub fn target_nodes<'b>(
+    key: Nibbles,
+    value: Option<Vec<u8>>,
+    proof: impl IntoIterator<Item = (&'b Nibbles, &'b Bytes)>,
+    mut witness: Option<&mut HashMap<B256, Bytes>>,
```

Comment:
pedantic nit/optional suggestion: I'd like to swap positions here
prefer having iterators last because this is usually more readable on the callsite

Reply:
sure done here https://github.com/paradigmxyz/reth/pull/12510/commits/013968ce6ea2a05044d638337bc49bb0210b9229